bundle/API/Repository/TagsService.php

```diff
@@ -173,25 +173,40 @@ public function getTagSynonymCount(Tag $tag, array $languages = null, $useAlways
      * @param int $offset The start offset for paging
      * @param int $limit The number of content objects returned. If $limit = -1 all content objects starting at $offset are returned
      * @param bool $returnContentInfo
+     * @param array $sortClauses
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion[] $additionalCriteria Additional criteria for filtering related content
      *
-     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user is not allowed to read tags
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the specified tag is not found
+     * @throws \eZ\Publish\Core\Base\Exceptions\UnauthorizedException
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Content[]|\eZ\Publish\API\Repository\Values\Content\ContentInfo[]
      */
-    public function getRelatedContent(Tag $tag, $offset = 0, $limit = -1, $returnContentInfo = true);
+    public function getRelatedContent(Tag $tag, $offset = 0, $limit = -1, $returnContentInfo = true, array $sortClauses = [], array $additionalCriteria = []);
 
     /**
      * Returns the number of content objects related to $tag.
      *
      * @param \Netgen\TagsBundle\API\Repository\Values\Tags\Tag $tag
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion[] $additionalCriteria Additional criteria for filtering related content
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user is not allowed to read tags
      * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the specified tag is not found
      *
      * @return int
      */
-    public function getRelatedContentCount(Tag $tag);
+    public function getRelatedContentCount(Tag $tag, array $additionalCriteria = []);
+
+    /**
+     * Returns facets for given $facetBuilders,
+     * for content tagged with $tag.
+     *
+     * @param \Netgen\TagsBundle\API\Repository\Values\Tags\Tag $tag
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\FacetBuilder[] $facetBuilders
+     *
+     * @throws \Netgen\TagsBundle\Exception\FacetingNotSupportedException
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Search\Facet[]
+     */
+    public function getRelatedContentFacets(Tag $tag, array $facetBuilders = []);
```

Comment:
Maybe this should be extracted to a separate service, since this method is not really useful by itself to warrant it being placed in public API. What do you think?

Reply:
@emodric I was struggling with that as well but in the absence of a better idea I've implemented it in the easiest way. Maybe a separate `RelatedContentService` would do?

Reply:
Why not just `RelatedContentFacetBuilder` since it builds facets for related content search? `Service` has a different semantics in the domain.

Reply:
Fixed in https://github.com/netgen/TagsBundle/pull/98/commits/46986d8987d499d61b6cc3efb128e37fed405940.